app/assets/javascripts/channels/la_exercises.js

```diff
@@ -0,0 +1,250 @@
+$(document).on('turbolinks:load', function() {
+    if ($.isController('exercises') && $('.teacher_dashboard').isPresent()) {
+
+        const exercise_id = $('.teacher_dashboard').data().exerciseId;
+        const study_group_id = $('.teacher_dashboard').data().studyGroupId;
+
+        const specific_channel = { channel: "LaExercisesChannel", exercise_id: exercise_id, study_group_id: study_group_id };
+
+
+        App.la_exercise = App.cable.subscriptions.create(specific_channel, {
+            connected: function () {
+                // Called when the subscription is ready for use on the server
+            },
+
+            disconnected: function () {
+                // Called when the subscription has been terminated by the server
+            },
+
+            received: function (data) {
+                // Called when there's incoming data on the websocket for this channel
+                if (data.type === 'rfc') {
+                    handleNewRfCdata(data);
+                } else if (data.type === 'working_times') {
+                    handleWorkingTimeUpdate(data.working_time_data)
+                }
+            }
+        });
+
+        function handleNewRfCdata(data) {
+            let $row = $('tr[data-id="' + data.id + '"]');
+            if ($row.length === 0) {
+                $row = $($('#posted_rfcs')[0].insertRow(0));
+            }
+            $row = $row.replaceWithPush(data.html);
+            $row.find('time').timeago();
+            $row.click(function () {
+                Turbolinks.visit($(this).data("href"));
+            });
+        }
+
+        function handleWorkingTimeUpdate(data) {
+            const user_progress = data['user_progress'];
+            const additional_user_data = data['additional_user_data'];
+
+            const user = additional_user_data[additional_user_data.length - 1][0];
+            const position = userPosition[user.type + user.id]; // TODO validate: will result in undef. if not existent.
+            // TODO: Do update
```

Comment:
still open todos - postpone them for the meantime?

Reply:
Yes, there are still open ToDos. For now, new working time data is sent to the frontend but not handled yet. I still need to incorporate merging the new data with the existing graph.